### PR TITLE
organized the Element services away from main entry class

### PR DESCRIPTION
**Pull request by calebTree**

### CODE REVIEW

File: src/main/java/edu/metrostate/ics499/team2/services/ElementService.java [hunk 1]
```diff
@@ -37,5 +37,155 @@ public ElementService(ElementRepository elmRepo) {
...
+		elmRepo.save(new Element("Roentgenium",		"Rg", 	"Metal", 		111, (280), 0, 0));
+		elmRepo.save(new Element("Copernicium",		"Cn", 	"Metal", 		112, (285), 0, 0));
+		elmRepo.save(new Element("unknown", 		"Nh",  	"Metal", 		113, (284), 0, 0));
+		elmRepo.save(new Element("unknown", 		"Fl", 	"Metal", 		114, (289), 0, 0));
+		elmRepo.save(new Element("unknown", 		"Mc",  	"Metal", 		115, (289), 0, 0));
+		elmRepo.save(new Element("unknwon", 		"Lv", 	"Metal", 		116, (293), 0, 0));
+		elmRepo.save(new Element("unknown", 		"Ts", 	"Metal", 		117, (294), 0, 0));
+		elmRepo.save(new Element("unknown", 		"Og", 	"Nonmetal", 	118, (294), 0, 0));
+        System.out.println("Data creation complete...");
+    }
```
Comment (dnguyen000) [inline src/main/java/edu/metrostate/ics499/team2/services/ElementService.java:145]: Good call moving this out to a service. We want to keep the Main method as clean as possible.

**Comment by dnguyen000** [comment]
@calebTree looking at this PR again and had the following question: the implementation current here between service and controller seems a little duplicative. The service level should be where the business logic is held, whereas the controller handles the HTTP requests to the endpoint. That said, rather than instantiating the repo in the service as well, would it make more sense to instantiate the controller in the service and then handle those HTTP calls based on the business logic needs?

**Comment by calebTree** [comment]
> @calebTree looking at this PR again and had the following question: the implementation current here between service and controller seems a little duplicative. The service level should be where the business logic is held, whereas the controller handles the HTTP requests to the endpoint. That said, rather than instantiating the repo in the service as well, would it make more sense to instantiate the controller in the service and then handle those HTTP calls based on the business logic needs?

That sounds right, it is definitely not entirely thorough.

**Comment by calebTree** [comment]
To me it might actually make more sense to remove the redundancy of the repository instantiation by instantiating it in _only_ the service?

**Comment by calebTree** [comment]
In that configuration the controller connects to the service that connects to the repository that provides the data involved in the business logic. Does that work?

**Comment by calebTree** [comment]
I do actually see what you mean.

**Comment by dnguyen000** [comment]
So I think we will want [this](https://github.com/dnguyen000/ICS-499/blob/854f52275fb6286f9a9ea858c6c33169f21ea4ca/src/main/java/edu/metrostate/ics499/team2/services/ElementService.java#L20-L23) to be changed from instantiating the repo to the controller. From a design perspective, we will want only connection to the repo, and all interactions to the DB should be done through the controller. Hopefully this makes sense, let me know if you have questions/want to connect on this.

**Comment by calebTree** [comment]
Run a `git pull --rebase`. I reverted my last commit. I believe I have what we're talking about in [9958556](https://github.com/dnguyen000/ICS-499/pull/13/commits/99585562075c160991f20a6efe6db78348c22492).

**Comment by calebTree** [comment]
Would the `ElementController` now instantiate the `ElementService`?

**Review by atvick21 — APPROVED** [review]

**Review by f947 — APPROVED** [review]

**Comment by wale0** [comment]
good

**Review by wale0 — APPROVED** [review]

**Pull request merged**